plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -208,7 +208,15 @@ public static function get_note_from_db( $note_from_db ) {
 
 			$note_from_db->set_title( $locale->title );
 			$note_from_db->set_content( $locale->content );
-			$note_from_db->set_actions( SpecRunner::get_actions( $spec ) );
+
+			$localized_actions = SpecRunner::get_actions( $spec );
+			$actions_from_db   = $note_from_db->get_actions();
+			$actions_length    = count( $localized_actions );
+			// Manually copy the action id from the db to the localized action, since they were not being provided.
+			for ( $i = 0; $i < $actions_length; $i++ ) {
+				$localized_actions[ $i ]->id = $actions_from_db[ $i ]->id;
```

Comment:
I suppose this will do for now, it is a bit of a dirty hack, and there is no way to know if the order of the actions are the same.
I would instead create a `get_action_by_name` function and add the `id` based on that.
Something like this: `$localized_actions[ $i ]->id = Notes::get_action_by_name( $note_from_db, $localized_actions[ $i ]->name )->id` ( if the action exists of-course )